CodePush.js

```diff
@@ -41,7 +40,7 @@ async function checkForUpdate(deploymentKey = null) {
    * in any updates for current app store version, regardless of hash.
    */
   let queryPackage;
-  if (localPackage && localPackage.appVersion && semver.compare(localPackage.appVersion, config.appVersion) === 0) {
+  if (localPackage && localPackage.appVersion && localPackage.appVersion === config.appVersion) {
```

Comment:
I actually wonder whether we shouldn't just remove the second and third conditions entirely, and rely on the native side to determine whether a local package is still valid or not. I believe we had this check in back before we started clearing updates on disk, so it's potentially just legacy cruft that we don't need anymore. I don't believe that the Cordova plugin does any validation like this before using a local package to query the server with.